From 123b6eb5b2dbf3e3a635ee4d97c298748ac927aa Mon Sep 17 00:00:00 2001
From: Gustavo Ornaghi Antunes <gustavo.ornaghiantunes@windriver.com>
Date: Wed, 24 Apr 2024 10:23:41 -0300
Subject: [PATCH] csi-powermax: add snapshotclass in templates

This patch consists of adding a snapshot class in the helm chart
templates directory.
It's needed because this helm chart needs a snapshot classes that should
be created manually.

Signed-off-by: Gustavo Ornaghi Antunes <gustavo.ornaghiantunes@windriver.com>
---
 charts/csi-powermax/templates/snapshotclass.yaml | 6 ++++++
 charts/csi-powermax/values.yaml                  | 4 ++++
 2 files changed, 10 insertions(+)
 create mode 100644 charts/csi-powermax/templates/snapshotclass.yaml

diff --git a/charts/csi-powermax/templates/snapshotclass.yaml b/charts/csi-powermax/templates/snapshotclass.yaml
new file mode 100644
index 0000000..f1a0ed6
--- /dev/null
+++ b/charts/csi-powermax/templates/snapshotclass.yaml
@@ -0,0 +1,6 @@
+apiVersion: snapshot.storage.k8s.io/v1
+kind: VolumeSnapshotClass
+metadata:
+  name: {{ .Release.Name }}-snapshotclass
+driver: {{ .Values.driverName | default "csi-powermax.dellemc.com" }}
+deletionPolicy: {{ .Values.controller.snapshot.deletionPolicy }}
diff --git a/charts/csi-powermax/values.yaml b/charts/csi-powermax/values.yaml
index 3b3e4c0..7932c3e 100644
--- a/charts/csi-powermax/values.yaml
+++ b/charts/csi-powermax/values.yaml
@@ -65,6 +65,8 @@ csm:
   replication: false
   authorization: false
 
+driverName:
+
 driver:
   # image: Image used for driver container. This shouldn't be changed
   # Allowed values: string
@@ -164,6 +166,8 @@ controller:
     # Default value: registry.k8s.io/sig-storage/csi-snapshotter:v7.0.1
     image: registry.k8s.io/sig-storage/csi-snapshotter:v7.0.1
 
+    deletionPolicy: Delete
+
   resizer:
     # enabled: Enable/Disable volume expansion feature
     # Allowed values:
-- 
2.34.1
